cunumeric/array.py

```diff
@@ -2464,6 +2456,59 @@ def diagonal(
                 raise ValueError("Either axis1/axis2 or axes must be supplied")
         return self._diag_helper(offset=offset, axes=axes, extract=extract)
 
+    @add_boilerplate("indices", "values")
+    def put(
+        self, indices: ndarray, values: ndarray, mode: str = "raise"
+    ) -> None:
+        """
+        Replaces specified elements of the array with given values.
+
+        Refer to :func:`cunumeric.put` for full documentation.
+
+        See Also
+        --------
+        cunumeric.put : equivalent function
+
+        Availability
+        --------
+        Multiple GPUs, Multiple CPUs
+
+        """
+
+        if values.size == 0 or indices.size == 0:
+            return
+
+        if mode not in ("raise", "wrap", "clip"):
+            raise ValueError(
+                "mode must be one of 'clip', 'raise', or 'wrap' "
+                f"(got  {mode})"
+            )
+
+        if mode == "wrap":
+            indices = indices % self.size
+        elif mode == "clip":
+            indices = indices.clip(0, self.size - 1)
+
+        indices = indices._warn_and_convert(np.dtype(np.int64))
+
+        if indices.ndim > 1:
+            indices = indices.ravel()
+
+        # in case if there are repeated entries in the indices array
+        # Legate doesn't guarantee the order in which values
+        # are updated
+        if indices.size > indices.unique().size:
+            runtime.warn(
+                "size of indices is larger than source array which"
+                " might yield results different from NumPy",
+                category=RuntimeWarning,
+            )
```

Comment:
This is an expensive check, and I wouldn't like to have it happen indiscriminately. I think it's sufficient to note this on the documentation for `put`.

Reply:
Sure, we can do that. I was also thinking of checking this only in the debug mode, but, for this, we will have to have a `--debug` flag implemented for the python (or something like this)

Reply:
Yes, let's discuss this at the next meeting. We already have two "debugging mode" flags (`LEGATE_TEST` and `-cunumeric:test`), and we're talking about introducing another one, so I'd like to get everyone's opinion before we go and do this.

For now let's just move this warning to the documentation on `cunumeric.put`.

Reply:
sounds good